sabnzbd/filesystem.py

```diff
@@ -59,18 +59,37 @@ def get_ext(filename: str) -> str:
         return ""
 
 
+def is_listed_ext(ext: str, ext_list: list) -> bool:
+    """Check if the extension is listed. In case of a regexp the entire extension must be matched;
+    partial matches aren't accepted (e.g. 'r[0-9]{2}' will be treated the same as '^r[0-9]{2}$' and
+    thus return false for extentions such as 'r007' despite the substring match on 'r00').
+    """
+    for item in ext_list:
+        RE_EXT = sabnzbd.rss.convert_filter(item)
```

Comment:
Let's move it to `misc`, since we use it in more spots now.
After that we can merge.